ext2/ext2.c

```diff
@@ -645,29 +630,15 @@ static int _get_blkno(ext2_t* ext2, uint32_t* blkno)
 
         /* Scan the bitmap, looking for free bit */
         {
-            const uint64_t* p = (const uint64_t*)locals->bitmap.data;
-
-            /* skip over full long words */
-            {
-                const uint64_t x = 0xffffffffffffffff;
-                size_t n = locals->bitmap.size / sizeof(uint64_t);
-                size_t i = n;
+            const uint8_t* p;
 
-                while (n > 4 && p[0] == x && p[1] == x && p[2] == x &&
-                       p[3] == x)
-                {
-                    p += 4;
-                    i -= 4;
-                }
+            /* skip over full (0xff) bytes */
+            p = myst_memcchr(locals->bitmap.data, 0xff, locals->bitmap.size);
 
-                while (i && p[0] == x)
-                {
-                    p++;
-                    i--;
-                }
-
-                lblkno = (8 * sizeof(uint64_t)) * (n - i);
-            }
+            if (p)
+                lblkno = (p - locals->bitmap.data) * 8;
+            else
+                lblkno = locals->bitmap.size * 8;
```

Comment:
Shouldn't we break out here?

Reply:
Fixed. It works either way but your way is easier to read.

Reply:
It turns out is is a continue (to advance to the next group). :)